pandas/tests/io/parser/test_read_fwf.py

```diff
@@ -556,7 +556,7 @@ def test_variable_width_unicode():
     tm.assert_frame_equal(result, expected)
 
 
-@pytest.mark.parametrize("dtype", [{}, {"a": "float64", "b": str, "c": "int32"}])
+@pytest.mark.parametrize("dtype", [{}, {"a": "float64", "b": str, "c": "float16"}])
```

Comment:
Why did this change?

Reply:
The test failed due to the changes in this PR. I had put a (probably too short) remark in [my comment above](https://github.com/pandas-dev/pandas/pull/47168#issuecomment-1193404635): I actually extended the scope of this PR: Similar to int overflow handling (which was my original target), we also have non-unique behavior concerning lossy coercion of float to int between parsing engines and non-/extension dtypes: It raise if `engine="c"` or if we use an extension dtype, but it performs silent lossy coercion if `engine="python"` and a non-extension integer dtype is given. I'd consider this to be an unwanted inconsistency and unsafe behavior in general. The test however assumed lossy coercion explicitly and effectively uses `engine="python"` and a non-extension integer dtype. As far as I can see, the documentation does not specify the behavior for this situation, so it should be okay to change the test.